### ISSUE TITLE
Progress Report

### ISSUE THREAD

**Issue by ClashLuke** [ClashLuke/JanNet#34]

**Comment by ClashLuke** [comment]
Below you can see the initial model, using group linear layer (orange) and their all-attention counterpart with the parameters (red), when trained on video and language at the same time. 
![image](https://user-images.githubusercontent.com/39779310/104439611-94343380-5591-11eb-8e7f-c88e6e66e9a1.png)
For comparison, here is the same configuration, but trained without language token in linear scale
![image](https://user-images.githubusercontent.com/39779310/104440015-1d4b6a80-5592-11eb-9a54-a795a870ffcf.png)
..and in log scale
![image](https://user-images.githubusercontent.com/39779310/104440183-54218080-5592-11eb-89ae-fad54ece9cdb.png)

**Comment by ClashLuke** [comment]
With #40, we're starting to see great convergence on ThePile. The model starts to get highly unstable once it's starting to converge, but this is most likely due to a far too low weight decay.
![image](https://user-images.githubusercontent.com/39779310/105576352-72e1fd00-5d72-11eb-9f33-3c5179982352.png)
1e-2 and 3e-2 will be done next.

Note that this is done using a 5.5B model trained on one v3-8.

**Comment by ClashLuke** [comment]
After incorporating #51, which contains a fix in the attention mask that renders all previous experiments worthless, the BFP16 loss curves look like this
![Screenshot_20210209-191543_Ungoogled_Chromium_Extensions_1_1](https://user-images.githubusercontent.com/39779310/107409715-c1322280-6b0c-11eb-8afd-7e7620bb3059.png)
Where the red curve uses the Adam optimizer, the green curve uses SM3. Otherwise, both use the identical 2.7B model with a batch size of 64 and fp32 calculation with bfp16 storage.
